src/utilities/EventEmitter.ts

```diff
@@ -231,96 +238,60 @@ export default class EventEmitter {
 
   /**
    * Listen for a single occurrence of an event
-   * @param event the name of the event to listen to
+   * @param listenerOrEvents (optional) the name of the event to listen to
+   * @param listener (optional) the listener to be called
    */
-  once(event: string): Promise<void>;
-
-  /**
-   * Listen for a single occurrence of any event
-   * @param listener the listener to be called
-   */
-  once(listener: Function): void;
-
-  /**
-   * Listen for a single occurrence of an event
-   * @param event the name of the event to listen to
-   * @param listener the listener to be called
-   */
-  once(event?: string | string[] | null, listener?: Function): void;
-
-  once(...args: unknown[]): void | Promise<void> {
-    const argCount = args.length;
-    if ((argCount === 0 || (argCount === 1 && typeof args[0] !== 'function')) && Promise) {
-      const event = args[0];
+  once<K extends EventKey<T>>(
+    listenerOrEvents?: K | K[] | EventListener<T[K]>,
+    listener?: EventListener<T[K]>
+  ): void | Promise<any> {
+    // .once()
+    if ([listenerOrEvents, listener].filter((i) => i).length === 0) {
       return new Promise((resolve) => {
```

Comment:
Why does this only return a promise here and not for any other time of argument? I know this is a carry-over from ably-js, but is it behavior that people would expect? I can see someone trying to await `.once("event")` thinking it will wait until the event fires.

Reply:
That's on me. I've changed some behaviour that, I thought, was unnecessary, but not actually thought through this one before handover. I think your guess is spot on, but I wonder why is not in `.on`. Honestly, I'd probably remove it, I think this EventEmitter should primarily serve our use case, and we should make the API as clear as possible so any potential future consolidations are easier.